sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/AliasHelper.scala

```diff
@@ -98,10 +98,15 @@ trait AliasHelper {
   protected def trimNonTopLevelAliases[T <: Expression](e: T): T = {
     val res = e match {
       case a: Alias =>
-        val metadata = if (a.metadata == Metadata.empty) {
-          None
-        } else {
-          Some(a.metadata)
+        val metadata = a.explicitMetadata match {
+          // SPARK-39838 Preserve explicit metadata, even if it is empty
+          case Some(metadata) => Some(metadata)
+          case None =>
+            if (a.metadata == Metadata.empty) {
+              None
+            } else {
+              Some(a.metadata)
+            }
```

Comment:
```suggestion
        val metadata = if (a.metadata == Metadata.empty && a.explicitMetadata.isEmpty) {
          None
        } else {
          Some(a.metadata)
```

Reply:
Thanks, your suggestion indeed is much simplier to read and understand :)